Should be able to get the images if the engine is imported.

```
akosgarai@akos-szemetlada ~/Documents/golang/src/github.com/akosgarai/opengl_playground (deattach-engine*) $ go run examples/08-multiple-light/app.go
OpenGL version 4.1.0 NVIDIA 440.82
panic: open /home/akosgarai/Documents/golang/src/github.com/akosgarai/opengl_playground/examples/08-multiple-light/assets/metal.jpg: no such file or directory

goroutine 1 [running, locked to thread]:
github.com/akosgarai/playground_engine/pkg/texture.(*Textures).AddTexture(0xc0000c1278, 0xc0000e8000, 0x77, 0x812f0000812f, 0x260100002601, 0x6448f6, 0x10, 0x683d80, 0xa21b08)
	/home/akosgarai/Documents/golang/pkg/mod/github.com/akosgarai/playground_engine@v0.0.0-20200611154221-10f4a9d15314/pkg/texture/texture.go:65 +0x6ad
github.com/akosgarai/playground_engine/pkg/model.NewTexturedStreetLamp(0xc0c000003ecccccd, 0x40c00000bfa66666, 0x683d80, 0xa21b08, 0x7000a82a0)
	/home/akosgarai/Documents/golang/pkg/mod/github.com/akosgarai/playground_engine@v0.0.0-20200611154221-10f4a9d15314/pkg/model/streetlamp.go:88 +0x126
main.TexturedStreetLamp(0xc0c000003ecccccd, 0xc0bfa66666, 0xc0000c1788)
	/home/akosgarai/Documents/golang/src/github.com/akosgarai/opengl_playground/examples/08-multiple-light/app.go:115 +0x5d
main.main()
	/home/akosgarai/Documents/golang/src/github.com/akosgarai/opengl_playground/examples/08-multiple-light/app.go:244 +0x1ccb
exit status 2
```

--- a/README.md
+++ b/README.md
@@ -2,6 +2,34 @@
 
 This engine is written for playing around with the opengle v4.x features. It contains basic primitives, supports camera, light sources, materials, textures, (wavefront) model import / export solutions.
 
+## CLI
+
+This package provides a cli tool for downloading the necessary assets or shaders for an application.
+
+### Usage
+
+First of all, You have to install the cli-application.
+
+```
+go get github.com/akosgarai/playground_engine
+cd $GOPATH/src/github.com/akosgarai/playground_engine
+go install
+```
+
+After this step, You should be able to use the cli-tool. It's easy to use, it gets a command and a parameter as arguments. Without arguments it prints out the help menu. The most important command is the install, that You can use for downloading model textures or shaders.
+
+The following command creates a `shaders` directory (if not exists) and downloads the shader apps into the `shaders` dir.
+
+```
+playground_engine install shaders
+```
+
+This command creates an `assets` directory (if not exists) and downloads the assest files into this directory.
+
+```
+playground_engine install models
+```
+
 ![Sample gif from outer space](https://github.com/akosgarai/go_opengl_playground/blob/master/examples/07-textured-spheres/sample/sample.gif)
 
 ## Useful links

--- a/README.md
+++ b/README.md
@@ -30,6 +30,8 @@ This command creates an `assets` directory (if not exists) and downloads the ass
 playground_engine install models
 ```
 
+## Sample
+
 ![Sample gif from outer space](https://github.com/akosgarai/go_opengl_playground/blob/master/examples/07-textured-spheres/sample/sample.gif)
 
 ## Useful links

--- a/main.go
+++ b/main.go
@@ -2,6 +2,8 @@ package main
 
 import (
 	"fmt"
+	"io"
+	"net/http"
 	"os"
 )
 
@@ -32,15 +34,51 @@ func createTarget(targetPath string) {
 		os.MkdirAll(targetPath, os.ModePerm)
 	}
 }
+func fileExists(filepath string) bool {
+	if _, err := os.Stat(filepath); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
 func downloadShaders() {
 	fmt.Println("Downloading shaders ...")
+}
+func downloadFile(from, target string) error {
+	// create the file
+	to, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer to.Close()
+	// get the data from the given url
+	resp, err := http.Get(from)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// write  the downloaded data to the file
+	_, err = io.Copy(to, resp.Body)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 func downloadModels() {
 	createTarget(modelTargetDirectory)
 	fmt.Println("Downloading models ...")
 	for _, img := range modelImages {
+		target := modelTargetDirectory + "/" + img
+		if fileExists(target) {
+			continue
+		}
 		url := fmt.Sprintf("%s%s%s", downLoadBaseUrl, modelBaseDirectory, img)
 		fmt.Printf("Downloading '%s'\n", url)
+		err := downloadFile(url, target)
+		if err != nil {
+			fmt.Printf("Something happened during download: '%s'.\n", err.Error())
+			fmt.Println("Please try to install again.")
+		}
 	}
 }
 

--- a/main.go
+++ b/main.go
@@ -1,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	downLoadBaseUrl      = "https://raw.githubusercontent.com/akosgarai/playground_engine/master/"
+	shaderBaseDirectory  = ""
+	modelBaseDirectory   = "pkg/model/assets/"
+	modelTargetDirectory = "assets"
+)
+
+var (
+	commands = map[string][]string{
+		"help":    []string{},
+		"install": []string{"shaders", "models"},
+	}
+
+	modelImages = []string{
+		"concrete-wall.jpg",
+		"crystal-ball.png",
+		"door.jpg",
+		"metal.jpg",
+	}
+)
+
+func createTarget(targetPath string) {
+	if _, err := os.Stat(targetPath); os.IsNotExist(err) {
+		fmt.Printf("Creating '%s' directory.\n", targetPath)
+		os.MkdirAll(targetPath, os.ModePerm)
+	}
+}
+func downloadShaders() {
+	fmt.Println("Downloading shaders ...")
+}
+func downloadModels() {
+	createTarget(modelTargetDirectory)
+	fmt.Println("Downloading models ...")
+	for _, img := range modelImages {
+		url := fmt.Sprintf("%s%s%s", downLoadBaseUrl, modelBaseDirectory, img)
+		fmt.Printf("Downloading '%s'\n", url)
+	}
+}
+
+func printCommandNames() {
+	for cmd, _ := range commands {
+		fmt.Printf("\t%s\n", cmd)
+
+	}
+}
+func helpMenu() {
+	fmt.Println("Welcome in the Playground Engine.\n")
+	fmt.Println("This CLI tool helps You to get the necessary assets.")
+	fmt.Println("How to use it?\n")
+	fmt.Println("\tcmd [command] [params]\n")
+	fmt.Println("Commands:\n")
+	printCommandNames()
+	fmt.Println("")
+}
+func helpMenuInstall() {
+	fmt.Println("With the install tool, You can download the necessary stuff for your application.\n")
+	for _, param := range commands["install"] {
+		fmt.Printf("If You want to download the %s, use the following command:\n\n", param)
+		fmt.Printf("\tplayground_engine install %s\n\n", param)
+	}
+}
+func installCommandHandler(args []string) {
+	if len(args) == 0 {
+		helpMenuInstall()
+		return
+	}
+	param := args[0]
+	switch param {
+	case "help":
+		break
+	case "shaders":
+		downloadShaders()
+		break
+	case "models":
+		downloadModels()
+		break
+	default:
+	}
+}
+
+func main() {
+	// Get the arguments without the program name
+	Args := os.Args[1:]
+	if len(Args) == 0 {
+		helpMenu()
+		return
+	}
+	Cmd := Args[0]
+	switch Cmd {
+	case "help":
+		helpMenu()
+		break
+	case "install":
+		installCommandHandler(Args[1:])
+		break
+	default:
+		fmt.Println("Unsupported command. Please use one of the followings:\n")
+		printCommandNames()
+		break
+	}
+
+}

--- a/main.go
+++ b/main.go
@@ -8,10 +8,11 @@ import (
 )
 
 const (
-	downLoadBaseUrl      = "https://raw.githubusercontent.com/akosgarai/playground_engine/master/"
-	shaderBaseDirectory  = ""
-	modelBaseDirectory   = "pkg/model/assets/"
-	modelTargetDirectory = "assets"
+	downLoadBaseUrl       = "https://raw.githubusercontent.com/akosgarai/playground_engine/master/"
+	shaderBaseDirectory   = "pkg/shader/apps/"
+	modelBaseDirectory    = "pkg/model/assets/"
+	modelTargetDirectory  = "assets"
+	shaderTargetDirectory = "shaders"
 )
 
 var (
@@ -25,6 +26,18 @@ var (
 		"crystal-ball.png",
 		"door.jpg",
 		"metal.jpg",
+	}
+	shaderApps = []string{
+		"material.frag",
+		"material.vert",
+		"point.frag",
+		"point.vert",
+		"texturecolor.frag",
+		"texturecolor.vert",
+		"texture.frag",
+		"texturemat.frag",
+		"texturemat.vert",
+		"texture.vert",
 	}
 )
 
@@ -41,7 +54,21 @@ func fileExists(filepath string) bool {
 	return true
 }
 func downloadShaders() {
+	createTarget(shaderTargetDirectory)
 	fmt.Println("Downloading shaders ...")
+	for _, img := range shaderApps {
+		target := shaderTargetDirectory + "/" + img
+		if fileExists(target) {
+			continue
+		}
+		url := fmt.Sprintf("%s%s%s", downLoadBaseUrl, shaderBaseDirectory, img)
+		fmt.Printf("Downloading '%s'\n", url)
+		err := downloadFile(url, target)
+		if err != nil {
+			fmt.Printf("Something happened during download: '%s'.\n", err.Error())
+			fmt.Println("Please try to install again.")
+		}
+	}
 }
 func downloadFile(from, target string) error {
 	// create the file

